docs/articles/documentation/test-api/handling-native-dialogs.md

```diff
@@ -0,0 +1,137 @@
+`function` | Function &#124; ClientFunction | A regular or [client function](executing-code-in-the-browser/client-functions.md) that will be triggered whenever a native dialog is invoked.
+`options`  | Object                         | See [Client Function Options](executing-code-in-the-browser/client-functions.md#options).
+
+The `function` has the following arguments.
+
+Argument | Type   | Description
+-------- | ------ | -------------
+`type`   | String | The type of the native dialog. `'alert' | 'confirm' | 'beforeunload' | 'prompt'`.
+`text`   | String | Text of the dialog message.
+
+Once the handler is specified, it will be triggered each time a native dialog appears in the test.
+You can provide a new handler at any moment by calling `setNativeDialogHandler` once again.
+If a native dialog appears when no handler is set, the test fails with an error.
+
+> To handle native dialogs that appear during the page load, specify the dialog handler
+> before the first test action.
+
+You can control how a dialog is handled via the handler's return values.
+If you return nothing, TestCafe performs default handling.
+
+What kind of values should be returned depends on the dialog type. See the table below for reference.
+
+Dialog Type  | Return Value                                             | Default Handling
+------------ | -------------------------------------------------------- | --------------
+alert        | Ignored                                                  | OK button.
+beforeunload | Ignored                                                  | Leave button. There is no way to emulate 'Stay' programmatically.
+confirm      | `true` to answer 'Yes'; `false` to answer 'No'.          | No button.
+prompt       | A string that contains text to be typed into the prompt. | Cancel button.
+
+The following example shows how to handle an alert dialog.
+
+```js
+test('My test', async t => {
+    await t
+        .setNativeDialogHandler(() => true)
+        .click('#show-alert-button');
+});
+```
+
+The next example demonstrates a test that handles two confirm dialogs and an alert dialog.
+
+```js
+test('My test', async t => {
+    await t
+        .setNativeDialogHandler((type, text) => {
+            if (type === 'confirm') {
+                switch (text) {
+                    case 'Do you want to subscribe?':
+                        return false;
+                        break;
+                    case 'Do you want to delete your account?':
+                        return true;
+                        break;
+                }
+            }
+            return true;
+        })
+        .click('#confirm-subscription');
+        .click('#show-alert');
+        .click('#confirm-account-deletion');
+});
+```
+
+## Dialog History
+
+You can get the history of invoked native dialogs to check if a certain dialog appeared or not.
+
+Use the `getNativeDialogHistory` function for this.
+
+```text
+getNativeDialogHistory() → [{type, text}]
```

Comment:
test controller's api looks great

```text
t.click(selector)
```

```text
t.getNativeDialogHistory() → [{type, text}]
```

But do you like this (like in MDN):

```text
Selector.prototype.with(options) → Selector
```

Because 

```text
Selector.with(options) → Selector
```

would mean it's static and

```text
selector.with(options) → Selector
```

doesn't make a big difference.

I like

```text
Selector.prototype.with(options) → Selector
```

but, on the other hand, we don't write

```text
TestController.prototype.click(selector)
```

@inikulin